### DISCUSSION TITLE
Migrate from CC2538 + CC2592 to SONOFF Zigbee 3.0 USB Dongle Plus

### DISCUSSION THREAD

**Discussion by jeppper** [Koenkk/zigbee2mqtt#10383] [Devices]
I have just received my new Sonoff Dongle but I am not quite sure how much work has to be done to do the swap from the old CC2538 + CC2592 coordinator.
Someone knows exactly what is needed to do this?
Does the new coordinator have to be flashed, upgraded and so on or is just plug´n play?
Will the new Sonoff work with Z2M Version 1.18.1?

**Comment by Koenkk** [comment]
https://www.zigbee2mqtt.io/guide/faq/#how-do-i-migrate-from-one-adapter-to-another

**Comment by jeppper** [comment]
Okay I have written the new IEEE adress in the adapter but what settings needs to be done in the Flash Programmer 2 when flashing the firmware. When selecting ERASE and PROGRAM it seems that the IEEE adress is set to default again.

![Unavngivet](https://user-images.githubusercontent.com/26521666/147472398-a2f19c40-21a6-4afc-aa9a-4c0a047a3d97.png)

**Reply by Koenkk** [reply]
See https://www.zigbee2mqtt.io/guide/adapters/flashing/flashing_via_uniflash.html

**Reply by jeppper** [reply]
I am not quite sure what to do with that. Can´t I use Flash Programmer 2 to flash the firmware after setting the IEEE adress?

**Reply by Koenkk** [reply]
Ah sorry, that is uniflash indeed. Your current settings looks good, now check Erase, Program and Verify and press the play button to flash

**Reply by jeppper** [reply]
I tried that but afterwards when trying to READ the IEEE adress it is FF FF FF  FF  FF FF FF FF again.
Should I maybe check the box at the right "RETAIN SECONDARY IEEE ADRESS" before flashing?

**Reply by Koenkk** [reply]
> Should I maybe check the box at the right "RETAIN SECONDARY IEEE ADRESS" before flashing?

yes you can try

**Reply by jeppper** [reply]
then the IEEE is retained :-)

**Reply by jeppper** [reply]
The new Sonoff Dongle is now up running but I had to repair most of my Xiaomi sensors anyway.

**Comment by jeppper** [comment]
Can I assume that the seriel port is the same as the old one "/dev/ttyACM0"?

**Reply by jeppper** [reply]
Actually I found out that USB0 was the new one.

**Comment by kirelam** [comment]
I have the issue, that the flash programmer V2 wil not change the IEEE adress of the sonoff zigbee 3.0 stick. I user the uartlog.py to set it in boot mode. After that I set the IEEE adress of the old stick in the secondary Mac adress field (in the mac tab) and click write. Nonetheless the IEEE adress does not change. I Will try uniflash tomorrow and post the results

**Reply by kirelam** [reply]
Die not try uniflash het, but flash programmer Will not change the main IEEE adress whatever i do. I set the old IEEE adress in the secondary field and ticket retain. After that nothing changed... Someone has any idea what I am doing wrong?

**Reply by jeppper** [reply]
I did not get the py file to work so pressed the "boot" button on the stick  instead. 
When doing so writing a new IEEE adress seems possible. 
If I read the adress afterwards the new adress shows up.

**Comment by artigbzh** [comment]
Hi. I got this usb bridge and install zigbee2mqtt web services on a debian server. Everything seems to work, except zigbee2mqtt detected nothing. So I suppose I need to update the firmware but the problem is I can't create account of  ti.com website (maintenance for accounts management)  to download Flash Programmer 2 . And it seems not possible to find it to another place on Internet.
Do you know how to find this tool ? 
Thanks

**Reply by oh2xx** [reply]
I had the same issue with the ti.com site so in the end I followed the [cc2538-bsl](https://www.zigbee2mqtt.io/guide/adapters/flashing/copy_ieeaddr.html) guide. The [video](https://www.youtube.com/watch?v=4S_c_m6z-RY&list=LL&index=1&t=7s) by digiblurDIY was also very helpful. The python update starts at the 10:29 mark.

**Comment by jeppper** [comment]
Unfortuenately I only know the ti site.
